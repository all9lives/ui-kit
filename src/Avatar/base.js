import { css } from 'styled-components'
import React, { Component } from 'react'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'
import View from '../View'
import Image from '../Image'
import Text from '../Text'

const SIZE_MAPPING = {
  small: 2,
  regular: 3,
  large: 4
}

export const Label = View.extend`
  position: absolute;
  border-radius: ${Unit.parse(3)};
  padding: ${Unit.parse(Size.spacing * 0.3)};
  background-color: ${Color.primary_light};
  right: ${Unit.parse(Size.spacing * 0.1)};
  bottom: ${props => props.name ? Unit.parse(Size.spacing * 2) : Unit.parse(Size.spacing * 0.1)};
`

export const LabelText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export const Name = View.extend`
  margin-top: ${Unit.parse(Size.spacing * 0.5)};
  background-color: ${Color.gray};
  border-radius: ${Unit.parse(Size.small)};
  align-items: center;
  justify-content: center;
`

export const NameText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export class Base extends Component {
  static defaultProps = {
    source: require('../assets/avatar-cat.jpg')
  }
  renderLabel = ({ label, name, labelStyle, labelTextStyle }) => (
    <Label name={name} style={labelStyle}>
      <LabelText style={labelTextStyle}>{label}</LabelText>
    </Label>
  )
  renderName = ({ name, nameStyle, nameTextStyle }) => (
    <Name style={nameStyle}>
      <NameText style={nameTextStyle}>{name}</NameText>
    </Name>
  )
  render () {
    const { size, source, name, label, borderRadius } = this.props
    return (
      <View direction='column' {...this.props}>
        <Image size={Size.icon * SIZE_MAPPING[size]} source={source} style={{ borderRadius }} />
        {name && this.renderName(this.props)}
        {label && this.renderLabel(this.props)}
      </View>
    )
  }
}

const base = css`
  position: relative;
`

export default base
