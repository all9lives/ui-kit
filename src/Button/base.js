import { css } from 'styled-components'
import Color from '../utils/Color'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export const attrs = {
  color: props => props.primary
    ? Color.primary
    : (Color[props.color] || props.color || Color.black)
}

const base = css`
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
`

export default base
