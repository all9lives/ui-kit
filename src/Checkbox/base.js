import React, { PureComponent } from 'react'
import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Spacing from '../utils/Spacing'
import Text from '../Text'

const base = css`
  position: relative;
  margin-right: ${Unit.convert(Spacing.regular)};
`

export const getComponent = ({ View, Checkbox }) => class Component extends PureComponent {
  handleValueChange = (value) => {
    if (this.props.onValueChange) {
      this.props.onValueChange(value)
    }
  }
  handleLabelPress = () => {
    if (this.props.onValueChange) {
      this.props.onValueChange(!this.props.value)
    }
  }
  render () {
    const { value, label } = this.props
    return (
      <View align='middle'>
        <Checkbox value={value} onValueChange={this.handleValueChange} />
        <Text alignSelf='center' onPress={this.handleLabelPress}>{label}</Text>
      </View>
    )
  }
}

export default base
