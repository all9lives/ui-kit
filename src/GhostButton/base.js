import React, { Component } from 'react'
import Unit from '../utils/Unit'
import Color from '../utils/Color'
import Size from '../utils/Size'
import View from '../View'
import Text from '../Text'

export const StyledView = View.extend`
  border-style: solid;
  border-radius: ${Unit.convert(Size.title)};
  border-width: 1px;
  border-color: ${props => Color[props.color] || props.color};
  padding: ${`${Unit.convert(Size.spacing * 1.5)} ${Unit.convert(Size.spacing * 5)}`};
`

export const StyledText = Text.extend`
  color: ${props => Color[props.color] || props.color};
  background-color: transparent;
`

export class Base extends Component {
  render () {
    const { title, ...props } = this.props
    return (
      <StyledView {...props}>
        <StyledText {...props}>
          {title}
        </StyledText>
      </StyledView>
    )
  }
}
