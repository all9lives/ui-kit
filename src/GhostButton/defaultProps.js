import Color from '../utils/Color'

export default {
  title: 'Button',
  color: Color.black
}
