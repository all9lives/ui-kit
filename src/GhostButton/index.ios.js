import GhostButton from './index.android'

export default GhostButton
