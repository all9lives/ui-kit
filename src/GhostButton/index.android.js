import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Base } from './base'
import defaultProps from './defaultProps'

const GhostButton = ({ onPress, ...props }) => (
  <TouchableOpacity onPress={onPress}>
    <Base {...props} />
  </TouchableOpacity>
)
GhostButton.defaultProps = defaultProps

export default GhostButton
