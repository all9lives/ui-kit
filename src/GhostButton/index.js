import React from 'react'
import { Base } from './base'
import defaultProps from './defaultProps'

const GhostButton = ({ onClick, onPress, ...props }) => (
  <span onClick={onClick || onPress}>
    <Base {...props} />
  </span>
)

GhostButton.defaultProps = defaultProps

export default GhostButton
