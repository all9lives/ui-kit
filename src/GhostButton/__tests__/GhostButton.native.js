/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import GhostButton from '../index.ios.js'
import { Base, StyledView, StyledText } from '../base'

const diveToView = wrapper => wrapper.find(Base).dive().find(StyledView).dive()
const diveToText = wrapper => wrapper.find(Base).dive().find(StyledText).dive()

describe('GhostButton.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <GhostButton>Apple</GhostButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `color:white`', () => {
    const wrapper = shallow(
      <GhostButton color='white'>Apple</GhostButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onPress`', () => {
    const wrapper = shallow(
      <GhostButton onPress={() => {}}>Apple</GhostButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `StyledView`', () => {
    const wrapper = shallow(
      <GhostButton />
    )
    expect(diveToView(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `StyledText`', () => {
    const wrapper = shallow(
      <GhostButton />
    )
    expect(diveToText(wrapper)).toMatchSnapshot()
  })
})
