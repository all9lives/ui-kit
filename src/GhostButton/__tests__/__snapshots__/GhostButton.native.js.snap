// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhostButton.native renders a snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Base
    color="#191919"
    title="Button"
  >
    Apple
  </Base>
</TouchableOpacity>
`;

exports[`GhostButton.native renders a snapshot with \`StyledText\` 1`] = `
<Text
  accessible={true}
  alignSelf="auto"
  allowFontScaling={true}
  color="#191919"
  disabled={false}
  ellipsizeMode="tail"
  margin={0}
  size={16}
  style={
    Array [
      Object {
        "alignSelf": "auto",
        "backgroundColor": "transparent",
        "color": "#191919",
        "fontSize": 16,
        "lineHeight": 20,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      },
      undefined,
    ]
  }
>
  Button
</Text>
`;

exports[`GhostButton.native renders a snapshot with \`StyledView\` 1`] = `
<View
  align=""
  color="#191919"
  direction="column"
  innerRef={[Function]}
  margin={0}
  padding={0}
  style={
    Array [
      Object {
        "borderColor": "#191919",
        "borderRadius": 25,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "column",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 15,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 15,
      },
      undefined,
    ]
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    Button
  </Styled(Text)>
</View>
`;

exports[`GhostButton.native renders a snapshot with \`color:white\` 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Base
    color="white"
    title="Button"
  >
    Apple
  </Base>
</TouchableOpacity>
`;

exports[`GhostButton.native renders a snapshot with \`onPress\` 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Base
    color="#191919"
    title="Button"
  >
    Apple
  </Base>
</TouchableOpacity>
`;
